larpandoracontent/LArTrackShowerId/MvaPfoCharacterisationAlgorithm.cc

```diff
@@ -383,20 +393,25 @@ StatusCode MvaPfoCharacterisationAlgorithm<T>::ReadSettings(const TiXmlHandle xm
         }
     }
 
+    // Need this in case we end up using the non-3d info version
     AlgorithmToolVector algorithmToolVector;
     PANDORA_RETURN_RESULT_IF(STATUS_CODE_SUCCESS, !=, XmlHelper::ProcessAlgorithmToolList(*this, xmlHandle, "FeatureTools", algorithmToolVector));
+    // and the map
+    LArMvaHelper::AlgorithmToolMap algorithmToolMap;
+    PANDORA_RETURN_RESULT_IF(STATUS_CODE_SUCCESS, !=, LArMvaHelper::ProcessAlgorithmToolListToMap(*this, xmlHandle, "FeatureTools", m_algorithmToolNames, algorithmToolMap));
```

Comment:
Well, partly tested. I was able to get the map output with a method like this, but running with 3D info off failed. But it also seemed to fail in a similar way if I just got our standard version and changed that one setting, so I'm not sure it's exactly related to this change...

Reply:
@AndyChappell do you expect this behavior or have a suggested better way to test?

Reply:
Actually it looks like in the 3D case you use the map, where as in the non 3D case you use the vector ([here](https://github.com/PandoraPFA/LArContent/blob/6322a22810d8cc9aa90d027ba5e83ac8d638d0d6/larpandoracontent/LArTrackShowerId/MvaPfoCharacterisationAlgorithm.cc#L403-L420)), presumably you would want to be able to support both here?

Reply:
To double check, do you mean I should remove the vector altogether and try to force the non-m_useThreeD version to also use the map? (I'm still not sure how to best test that that works if so)

Reply:
Attempted to do this in commit [b795d30](https://github.com/PandoraPFA/LArContent/pull/195/commits/b795d304739aecf45e0b0286661c7f6748028652) -- is this something you are able to test? (or could you point me to a version of the 2d feature training that I could test?)

There were also two ThreeD features for which I didn't make a version of `Run` using maps. Updated that in [23aa308](https://github.com/PandoraPFA/LArContent/pull/195/commits/23aa308e74a866c0d2f8eae78eb4182fb075ffec) but not tested.

Reply:
> To double check, do you mean...
Here I mean that whether or not the map or vector gets used depends on whether or not 3D info is being used, but that variable relates to whether or not you have charge information, right? So you would want to support the option of using the vector or the map in both the with and without charge info cases?

Reply:
> Attempted to do this in commit [b795d30](https://github.com/PandoraPFA/LArContent/pull/195/commits/b795d304739aecf45e0b0286661c7f6748028652) -- is this something you are able to test? (or could you point me to a version of the 2d feature training that I could test?)

This is determined by an XML parameter in the BdtPfoCharacterisation alg, specifically `<UseThreeDInformation>`, it appears to be set to true in the current configuration, so you can set this to false to test the alternative pathway.

Reply:
Ah maybe I misunderstood, I was thinking you meant remove the vector and make it use the Map even if `m_useThreeDInformation` is false, which is what my commit last night enforces.

Doesn't `m_useThreeDInformation` make it use the ClusterCharacterisation and not PfoCharacterisation? So I would need a BDT that is trained to use the TwoD vars and not the ThreeD vars, if I am thinking properly.

Reply:
In this context it will cause the PfoCharacterisation to load the weights file from the tree PfoCharBDTNoChargeInfo, in addition to the tree PfoCharBDT, both of which should be contained in PandoraBdt_v09_32_00_SBND.xml. It's perhaps not the best parameter name in this context, but essentially this no charge BDT would be used in the event that the W view is empty.